docs/developer/decision-records/2022-02-03-integration-testing/README.md

```diff
@@ -0,0 +1,301 @@
+# Integration testing
+
+## Decision
+
+Extend the existing `EdcExtension` JUnit facility (that can currently run a single EDC runtime and supports stubbing and extension of runtime services) as follows:
+
+- Load more than one EDC runtime.
+- Load EDC runtimes in separate Class Loaders.
+- Run each EDC runtime Class Loader with its effective runtime class path, based on its module's Gradle configuration.
+
+## Rationale
+
+The need to provide an integration test harness that supports multiple runtimes emerges from multiple needs:
+
+- Stabilizing [samples](../../samples) that run multiple connectors, which have been breaking frequently.
+- Testing system behavior when multiple connectors interact, e.g. the contract negotiation process.
+- Testing system behavior upon component failure.
+- Providing a test facility for factoring out application components to separate runtimes (e.g. [DPF](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/463)).
+
+Key drivers for the choice are:
+
+- Fast and efficient run in CI.
+- Fast "inner loop" (i.e. ability to quickly rerun tests after changing code) and debuggability for developers.
+- Use of existing frameworks, stability and portability.
+
+We have performed technical spikes testing multiple approaches (detailed further below), including various combinations of:
+
+- JUnit
+- [Docker compose](https://docs.docker.com/compose/)
+- Starting custom Class Loaders for separate threads for the Provider and Connector with separate class paths, and distinct instances of dependent instances, effectively providing full runtime isolation within a single JVM
```

Comment:
Not sure to follow what you mean with "distinct instances of dependent instances" here

Reply:
Rephrased